### ISSUE TITLE
GitJournal lost my file with no warning

### ISSUE THREAD

**Issue by FooBarWidget** [GitJournal/GitJournal#124]
I'm using GitJournal for the first time. I set up a repo on Github using the automatic setup.

I spent the past 2 hours writing a blog post. When it was finally done, I clicked on the sync button. And then... it deleted my file. No warning. No error. It's just gone. I checked my Git repo on Github, and nothing was committed here.

I made a new note, and tried to sync again. This time it said 'invalid credentials'. This *new* note is not deleted: only my first one was. I checked my Github repo, and saw that it had no deploy key. I'm not sure what happened to it: maybe I deleted it by mistake during an account settings cleanup?

I tried restarting GitJournal, but the file is still gone. I've lost the work.

I suspect that this data loss has got to do with not being able to push.

**Comment by vHanda** [comment]
Hey FooBar. This is terrible.

I really hope you haven't lost the file. I can do two things to prevent this in the feature -
* Any errors when saving the file, are displayed clearly and not in a scaffold.
* The contents are saved in some backup file as you're typing. So this way we should have the backup file, so you hopefully never loose data.

In order to help me diagnose this issue a bit better. Could you check your git log and see if you have a commit from that time or is there is even no commit?

**Comment by FooBarWidget** [comment]
Unfortunately, out of frustration, I uninstalled GitJournal. So any backup it might have is now gone. I could not find anything in the UI that mentions backups, so I assumed there is none.

My repo on Github shows no commits, other than the initial commit that adds a .gitignore.

I'm on Android. Samsung Galaxy S7.

**Comment by vHanda** [comment]
I completely understand.

Did you happen to add another note after that? Or do any other operations?

**Comment by FooBarWidget** [comment]
Yes I added another note after that. After adding the second note, I tapped the sync button. That sync failed, but did not delete the second note.

No other operations other than these.

**Comment by vHanda** [comment]
Last question: Just to confirm, did you see the note in the note list and then after it failed to sync it disappeared? Or you didn't even see it in the list?

**Comment by FooBarWidget** [comment]
I did see the note in the note list. Only after the sync did it disappear.

**Comment by vHanda** [comment]
Ok. Thanks. I understand.

I'll think tomorrow on what all checks I could do to make sure we never lose a note's content again. I'll keep you updated of what all I implement. But I completely understand if you rather not give GitJournal another chance. My apologies.

**Comment by joaosa** [comment]
Hi! Something similar to this has just happened to me on Android. 

It was an existing note and suddenly it was completely blank meaning I lost about one hour or so of changes. Is there any chance I can recover any of the changes? 
To my surprise they don't seem to have been committed and I was too late to check on my clipboard @vHanda 

Thank you!

**Comment by vHanda** [comment]
@joaosa : Shit. That's horrible. What do you mean by completely blank? As in all parts of GitJournal went blank? Including the nav bar and everything else?

Could you please send me the debug log of the app. `Sidebar -> Report a bug`. Also let's continue this conversation via email or another issue, please.

**Comment by joaosa** [comment]
> @joaosa : Shit. That's horrible. What do you mean by completely blank? As in all parts of GitJournal went blank? Including the nav bar and everything else?
> 
> Could you please send me the debug log of the app. `Sidebar -> Report a bug`. Also let's continue this conversation via email or another issue, please.

No worries! It wasn't a big loss thankfully. It was only the note itself that disappeared, i.e. the note is listed but its contents are blank.

I followed your instructions and reported the bug through the app. Let me know if you got it

**Comment by chrisalbrecht** [comment]
I just had a long running note that I've been working on for a long time disappear on android as well :(
